### PR TITLE
Wordnet embedding example

### PR DESCRIPTION
**Pull request by meiji163**
it's probably too long right now

### CODE REVIEW

File: hyperlib/models/pehr.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+from hyperlib.manifold.lorentz import Lorentz
+from hyperlib.manifold.poincare import Poincare
+from hyperlib.loss.constrastive_loss import contrastive_loss
+
+
+log = logging.getLogger(__name__)
+
+
+class HierarchicalEmbeddings(tf.keras.Model):
```
Comment (meiji163) [inline hyperlib/models/pehr.py:14]: We should do follow up work on the API for this in the future, for the purposes of the example it's good

File: hyperlib/loss/constrastive_loss.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import tensorflow as tf
+
+
+def contrastive_loss(pos_embs, neg_embs, M, c=1.0, clip_value=0.9):
```
Comment (meiji163) [inline hyperlib/loss/constrastive_loss.py:4]: There's also follow up work for this, mainly that it only works with `M=Poincare()` and expects `pos_emb`,`neg_emb` in a particular format that's not well documented.

**Review by meiji163 — COMMENTED** [review]
LGTM
